File: AlertToast/app/src/main/java/com/alfianyusufabdullah/alerttoast/MainActivity.java
package com.alfianyusufabdullah.alerttoast;

import android.support.design.widget.TextInputEditText;
import android.support.design.widget.TextInputLayout;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    Button btnShortToast, btnLongToast;
    TextInputEditText etText;
    TextInputLayout inputText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etText = findViewById(R.id.etText);

        inputText = findViewById(R.id.inputText);

        btnShortToast = findViewById(R.id.btnShortToast);
        btnShortToast.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String text = etText.getText().toString();

                if (text.isEmpty()) {
                    inputText.setError("Masukkan Text");
                    return;
                }

                hideSoftkey();

                Toast.makeText(MainActivity.this, text, Toast.LENGTH_SHORT).show();
                inputText.setErrorEnabled(false);

            }
        });

        btnLongToast = findViewById(R.id.btnLongToast);
        btnLongToast.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String text = etText.getText().toString();
                if (text.isEmpty()) {
                    inputText.setError("Masukkan Text");
                    return;
                }

                hideSoftkey();

                Toast.makeText(MainActivity.this, text, Toast.LENGTH_LONG).show();
                inputText.setErrorEnabled(false);

            }
        });
    }

    private void hideSoftkey() {
        InputMethodManager im = (InputMethodManager) getSystemService(INPUT_METHOD_SERVICE);
        if (im != null) {
            im.hideSoftInputFromWindow(etText.getWindowToken(), 0);
        }
    }
}

File: AlertToast/app-kotlin/src/main/java/com/alfianyusufabdullah/alerttoast_kotlin/MainActivity.kt
package com.alfianyusufabdullah.alerttoast_kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnShortToast.setOnClickListener {
            val text = etText.text.toString()

            if (text.isEmpty()) {
                inputText.error = "Masukkan Text"
                return@setOnClickListener
            }

            Toast.makeText(this, text, Toast.LENGTH_SHORT).show()
            inputText.isErrorEnabled = false

        }

        btnLongToast.setOnClickListener {
            val text = etText.text.toString()

            if (text.isEmpty()) {
                inputText.error = "Masukkan Text"
                return@setOnClickListener
            }

            Toast.makeText(this, text, Toast.LENGTH_LONG).show()
            inputText.isErrorEnabled = false
        }
    }
}
